### ISSUE TITLE
OpenGL issues team E-Mail address not working

### ISSUE THREAD

**Issue by mfleisz** [MicrosoftDocs/windows-driver-docs#2951]
Over at https://docs.microsoft.com/en-us/windows-hardware/drivers/display/loading-an-opengl-installable-client-driver it says:

> To obtain a license for the OpenGL ICD Development Kit, contact the OpenGL Issues team

referencing to the mail address opengl@microsoft.com

Whenever I send a mail to this address (I already tried with different mail accounts) I receive the following error:

```
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a permanent error. The following address(es) failed:

  opengl@microsoft.com
    host microsoft-com.mail.protection.outlook.com [104.47.53.36]
    SMTP error from remote mail server after RCPT TO:<opengl@microsoft.com>:
    550 5.4.1 Recipient address rejected: Access denied. AS(201806281) [BL2NAM06FT004.Eop-nam06.prod.protection.outlook.com]
```

Can you please provide the correct mail address to reach out to the OpenGL issues team or let them know that there is something wrong with their mail configuration?

Thanks, Martin

**Comment by lorihollasch** [comment]
Thanks for reporting this issue, @mfleisz . ICD is legacy, the dev team is no longer doing the kit, and the alias is defunct so I've removed references to it.

**Comment by mfleisz** [comment]
@lorihollasch Thanks for the heads-up! So if ICD is legacy what else is the preferred way for IHV to provide OpenGL support?

**Comment by lorihollasch** [comment]
@mfleisz , I reread the email from the dev team and see that my original comment was wrong, sorry! The ICD _kit_ has not existed for many many years. Everything needed to develop an ICD has been a part of the Windows SDK and WDK for several years. I've updated my original comment. Apologies again!

**Comment by mfleisz** [comment]
@lorihollasch thanks for the additional information!

However, are you sure that everything needed to develop an ICD is included in the Windows SDK/WDK? We have searched through all the header files and the only reference to OpenGL ICD we found is `D3DKMT_OPENGLINFO` in d3dkmthk.h.

Nothing about the ICD API or whatsoever. Could you give us a hint where to start?

**Comment by mfleisz** [comment]
I.e. the ICD exports functions like `DrvCreateContext` but searching for these strings yields no results.

**Comment by lorihollasch** [comment]
@mfleisz , I need to apologize again. It turns out that my original comment was correct so I've reinstated it. My recent exchange with the dev team seemed to suggest that only the alias was gone. However, I did more sleuthing and found that the ICD OpenGL driver model itself is indeed deep legacy. If you are going to continue down your path, have you looked at _gldrv.h_? It was originally provided to IHVs on request, but was added to the WDK a couple of years ago as a formality (it's in the _shared_ directory).
